File: VB/vbnet/vbnet/Get_Data/Retrieve_Transaction_Data.vb
' *
' * Bluepay VB.NET Sample code.
' *
' * This code sample runs a report that grabs data from the
' * BluePay gateway based on certain criteria. See comments below
' * on the details of the report.
' * If using TEST mode, only TEST transactions will be returned.
' *

Imports System
Imports vbnet.BPVB

Namespace GetData

    Public Class RetrieveTransactionData

        Public Shared Sub run()
            
            Dim accountID As String = "Merchant's Account ID Here"
            Dim secretKey As String = "Merchant's Secret Key Here"
            Dim mode As String = "TEST" 
            
            Dim query As BluePay = New BluePay(
                accountID,
                secretKey,
                mode
            )

          ' Report Stard Date: YYYY-MM-DD
          ' Report End Date: YYYY-MM-DD
          ' Also search subaccounts? Yes
          ' Output response without commas? Yes
          ' Do not include errored transactions? Yes
            query.getTransactionReport(
              reportStartDate:="2015-01-01", 
              reportEndDate:="2015-04-30", 
              queryByHierarchy:="1", 
              doNotEscape:="1", 
              excludeErrors:="1" 
            )
            
            query.process()

            Console.Write(query.response)

        End Sub
    End Class
End Namespace

File: C#/bluepay-c#/Get_Data/Retrieve_Transaction_Data.cs
/*
* BluePay C#.NET Sample code.
*
* This code sample runs a report that grabs data from the
* BluePay gateway based on certain criteria. See comments below
* on the details of the report.
* If using TEST mode, only TEST transactions will be returned.
*/

using System;
using BluePayLibrary;

namespace GetData
{
    class RetrieveTransactionData
    {
        public static void Main()
        {
            string accountID = "Merchant's Account ID Here";
            string secretKey = "Merchant's Secret Key Here";
            string mode = "TEST";
            
            BluePay report = new BluePay
            (
                accountID,
                secretKey,
                mode
            );

            report.GetTransactionReport
            (
                reportStartDate: "2015-01-01", // YYYY-MM-DD
                reportEndDate: "2015-05-30", // YYYY-MM-DD
                subaccountsSearched: "1", // Also search subaccounts? Yes
                doNotEscape: "1", // Output response without commas? Yes
                excludeErrors: "1" // Do not include errored transactions? Yes
            );

            // Makes the API request with BluePay
            report.Process();

            // Reads the reportesponse from BluePay
            Console.Write(report.response);
        }
    }
}